src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs

```diff
@@ -174,14 +174,13 @@ protected virtual void CleanUpCollectionViewSource()
 				CollectionViewSource.Source = null;
 				CollectionViewSource = null;
 			}
-			VirtualView?.ClearLogicalChildren();
 
-			// Add back header and footer as logical children. Otherwise they won't render
-			if (ListViewBase?.Header is View header)
-				VirtualView?.AddLogicalChild(header);
-
-			if (ListViewBase?.Footer is View footer)
-				VirtualView?.AddLogicalChild(footer);
+			// Remove all children inside the ItemsSource
+			foreach (var item in ListViewBase.GetChildren<ItemContentControl>())
+			{
+				var element = item.GetVisualElement();
+				VirtualView?.RemoveLogicalChild(element);
```

Comment:
If `VirtualView` is `null`, there's no reason to iterate over all the children. Can you check it once before iterating?

Reply:
Done